### PR TITLE
Reorg blk/isrt and configure in linter ppl

### PR DESCRIPTION
**Pull request by YunchuWang**
* Fix linter pipeline to use black/isort instead of flake8
* Create separate linter pipeline per project for better isolation
* Give pipeline write permission for reporting linting errors back to github.
* Change the ut pipelines to trigger on project itself only.

### CODE REVIEW

File: .github/workflows/ci_ut_ext_blob_workflow.yml [hunk 1]
```diff
@@ -4,12 +4,10 @@ on:
     push:
       branches: [ dev, master, main, release/* ]
       paths:
-        - 'azure-functions-extension-base/**'
         - 'azure-functions-extension-blob/**'
     pull_request:
         branches: [ dev, master, main, release/* ]
         paths:
-          - 'azure-functions-extension-base/**'
```
Comment (gavin-aguiar) [inline .github/workflows/ci_ut_ext_blob_workflow.yml:12]: Why is this removed?

Comment (gavin-aguiar) [reply .github/workflows/ci_ut_ext_blob_workflow.yml:12]: Can you update the description of this PR. Not sure whats happening here

File: .github/workflows/linting_extension_base.yml [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+
+      ################################
+      # Run Linter against code base #
+      ################################
+      - name: Lint Code Base
+        uses: super-linter/super-linter@v6.3.0
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          VALIDATE_PYTHON: true
+          # VALIDATE_PYTHON_FLAKE8: false
```
Comment (gavin-aguiar) [inline .github/workflows/linting_extension_base.yml]: Remove this if its not needed

File: .github/workflows/linting_extension_base.yml [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+          # VALIDATE_PYTHON_PYLINT: false # disable pylint, as we have not configure it
+          VALIDATE_PYTHON_BLACK: true # same as above
+          VALIDATE_PYTHON_ISORT: true # same as above
+          # PYTHON_BLACK_CONFIG_FILE: .isort.cfg
```
Comment (gavin-aguiar) [inline .github/workflows/linting_extension_base.yml]: Remove all the commented code

File: .github/workflows/ci_ut_ext_blob_workflow.yml (hunk 1 above)
Comment (YunchuWang) [reply .github/workflows/ci_ut_ext_blob_workflow.yml:12]: updated, as we discussed if we dont install editable base package in fastapi or blob pipeline for testing, there is no need to trigger blob or fastapi if base repo changes.

File: .github/workflows/ci_ut_ext_blob_workflow.yml [hunk 4]
```diff
@@ -4,12 +4,10 @@ on:
     push:
       branches: [ dev, master, main, release/* ]
       paths:
-        - 'azure-functions-extension-base/**'
         - 'azure-functions-extension-blob/**'
     pull_request:
         branches: [ dev, master, main, release/* ]
```
Comment (gavin-aguiar) [inline .github/workflows/ci_ut_ext_blob_workflow.yml]: Also remove master since we dont have that branch

**Review by gavin-aguiar — APPROVED** [review]

**Review by hallvictoria — APPROVED** [review]
